core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -123,7 +170,19 @@ public boolean canMerge(ScanTask other) {
     @Override
     public SplitScanTask merge(ScanTask other) {
       SplitScanTask that = (SplitScanTask) other;
-      return new SplitScanTask(offset, len + that.length(), fileScanTask);
+      return new SplitScanTask(offset, len + that.length(), fileScanTask, deletesSizeBytes);
```

Comment:
Should we use the method `deleteSizeBytes` instead of the variable `deleteSizeBytes`?

Reply:
We use the variable on purpose to avoid triggering the computation. Only use if already computed.

Reply:
I should have added a comment, will do tomorrow.

Reply:
```
SplitScanTask one = xxxx;
SplitScanTask two = xxx;
SplitScanTask three = one.merge(two); // Now the variable deleteSizeBytes is zero
three.deleteSizeBytes(); // we will compute the deleteSizeBytes
one.deleteSizeBytes();  // We will compute the deleteSizeBytes
```
It will recompute the deleteSizeBytes repeatedly in this situation. Could you point it out if I am wrong?

Reply:
When `SplitScanTask` is created from `BaseFileScanTask` during planning, we are using `deleteSizeBytes()` to trigger the computation once per `BaseFileScanTask` instead of once per `SplitScanTask` (the number of split tasks usually matches the number of row groups). This is because each `SplitScanTask` has the same set of deletes as the parent task it was created from. After planning, adjacent `SplitScanTask`s in the same bin are merged together. While merging we are using `deleteSizeBytes` variable (not method), which should be already populated if the split tasks were created from the parent task. If the variable is not populated, it means tasks were created via a separate process (like parsing). In that case, we don't know whether it is beneficial to compute `deleteSizeBytes` so we skip computing it unless requested.

Reply:
> When `SplitScanTask` is created from `BaseFileScanTask` during planning, we are using `deleteSizeBytes()` to trigger the computation once per `BaseFileScanTask` instead of once per `SplitScanTask` (the number of split tasks usually matches the number of row groups). This is because each `SplitScanTask` has the same set of deletes as the parent task it was created from. After planning, adjacent `SplitScanTask`s in the same bin are merged together. While merging we are using `deleteSizeBytes` variable (not method), which should be already populated if the split tasks were created from the parent task. If the variable is not populated, it means tasks were created via a separate process (like parsing). In that case, we don't know whether it is beneficial to compute the size of deletes so we skip computing it unless requested.

Thanks for your explanation.  I get your point.  Make sense.